### ISSUE TITLE
Error running test.lua

### ISSUE THREAD

**Issue by srgtm** [NorthernMan54/homebridge-wssensor#8]
```
opening port /dev/cu.usbserial-1430 with 115200 baud
Executing test.lua
dofile("test.lua")
Heap Available:  27968
Heap Available: config 29104
Heap Available: led 29104
Heap Available: setup 29200
Booting...
?:0: attempt to call field 'alarm' (a nil value)
stack traceback:
	?: in function '?'
	?: in function 'boot'
	test.lua:74: in main chunk
	[C]: in function 'dofile'
	stdin:1: in main chunk
>

**Comment by srgbpnkru** [comment]
```
opening port /dev/cu.usbserial-1430 with 115200 baud
Executing test.lua
dofile("test.lua")
Compiling:	config.lua
done
Heap Available:  28608
Heap Available: config 27104
Heap Available: led 24208
Heap Available: setup 22744
Booting...
?:0: attempt to call field 'alarm' (a nil value)
stack traceback:
	?: in function '?'
	?: in function 'boot'
	test.lua:74: in main chunk
	[C]: in function 'dofile'
	stdin:1: in main chunk
>

**Comment by srgbpnkru** [comment]
First is with ACL config, second with BME-MS.
Which one should I use?

**Comment by NorthernMan54** [comment]
What sensors do you have attached?

**Comment by srgbpnkru** [comment]
Sorry, haven't attached the temp/humidity sensor, thought that is optional. Will purchase it ASAP and try again.

**Comment by srgbpnkru** [comment]
I

**Comment by NorthernMan54** [comment]
Looking at this, the nodemcu firmware changed the api for the TMR function recently as my codebase doesn't didn't support that version.  I have updated the problematic package led.lua and it should be okay now.  Just install the latest version of led.lua and you should be good.

**Comment by srgbpnkru** [comment]
Awesome, thank you, it did the trick. But now it's stuck on `Configuring Wifi ...`. I've doublechecked the passwords.lua file – it contains proper information for sure.
```
opening port /dev/cu.usbserial-1430 with 115200 baud
Executing test.lua
dofile("test.lua")
Heap Available:  33848
Heap Available: config 32328
Heap Available: led 29040
Heap Available: setup 27736
Booting...
Configuring Wifi ...
>

**Comment by NorthernMan54** [comment]
This is the code it is executing

https://github.com/NorthernMan54/homebridge-wssensor/blob/master/nodemcu/lua/setup.lua

My guess is that is the passwords file, as you need to include the ssid twice.  

You may want to add a print statement after line 7, to validate the If statement on 8.  My guess is that the key isn’t finding the correct object in the passwords file.

**Comment by srgbpnkru** [comment]
Did it. Not even getting there. 

My passwords.lua is this:
```
-- Rename this to passwords.lua, and add your WIFI SSID and Password

local module = {}

module.SSID["Wifi2"] = { ssid="Wifi2", pwd = "12345678" }
module.SSID["Wifi5"] = { ssid="Wifi5", pwd = "12345678" }
return module
```
So I think it's correct. 

I've tried to put a print statement after line 3 in setup.lua to check weither it's getting there and it doesn't. Maybe `wifi.sta.getap()` fails to get list of APs? Maybe defunct Nodemcu?

**Comment by srgbpnkru** [comment]
Maybe it’s that: https://stackoverflow.com/questions/46896603/cant-connect-to-wifi-nor-wifi-sta-getap-doesnt-work
Will try it later today.

**Comment by srgbpnkru** [comment]
Hello, any updates?
Sensors and the WiFi connectivity tested — all working.

**Comment by NorthernMan54** [comment]
Where is it stuck?

**Comment by srgbpnkru** [comment]
At the setup.lua:30 I believe.

**Comment by NorthernMan54** [comment]
Defective board?

Basically if line 30 `wifi.sta.getap(0,wifi_start)` doesn't execute, it is an issue with the board receiving WIFI.  Incompatible WIFI or defective board or bad nodemcu firmware

**Comment by srgbpnkru** [comment]
I've tried different sketches and the Blynk app. I don't know they are working the same way as your's, but I've connected to the board through the wifi and I'm reading sensor data, so it's not that defective. I think it's a firmware, because all other times I was using Arduino IDE to flash and upload. So maybe that states that Arduino IDE firmware is OK, but online-generated is not? Or maybe it's the PyFlasher that I'm using? I will try CLI tool. And another image.
